### ISSUE TITLE
Update relationships

### ISSUE THREAD

**Issue by lu40** [vuex-orm/vuex-orm#238]
I was looking for a way to update relationships (like the [`attach` and `associate` methods](https://laravel.com/docs/5.7/eloquent-relationships#updating-many-to-many-relationships) and their counterparts used in Laravel's ORM) but couldn't find it. Is there any way to update relationships, else than inserting with related data or directly adding records to the pivot table (for the many-to-many case) I haven't found? Or is there anything similar planned?

Anyway thanks a lot for this great plugin! I've been using it together with Laravel and Nuxt.js and it plays together just perfect!

**Comment by kiaking** [comment]
Thanks for the feedback! No, sorry but there's no such methods at the moment.

I do think this is a really nice feature and definitely we should support it! It might take time since there're lots of another issue, but I would like to keep this issue open and close when the feature is implemented : )

**Comment by rkingon** [comment]
+1

I'm finding myself in need of this as well. I have a scenario where I `insert` without relations. Later on, I `update` with relations. Problem is those relations do not get `inserted`

```js
store.dispatch(`entities/${entity}/update`, {
  data: {
    ...updateSomeProps,
    someHasManyRelation: [
      { ... }, { ... }
    ]
  }
})
```

`someHasManyRelation` comes back empty

**Comment by kiaking** [comment]
@rkingon Hi! Thanks for the feedback. I too think this issue is important but your case is not related to this one.

Use `insertOrUpdate` instead of `update` in that case. The reason why relationship is not inserted is because it's trying to "update" the relationship as well. Since relationships doesn't exists, they're failing at updating. By using `insertOrUpdate`, the relationships should be inserted.

```js
store.dispatch(`entities/${entity}/insertOrUpdate`, {
  data: {
    ...updateSomeProps,
    someHasManyRelation: [
      { ... }, { ... }
    ]
  }
})
```

Or, you could explicitly specify how you would want to insert the relationships by specifying the options. Here I assume the relationship name is `posts`.

```js
store.dispatch(`entities/${entity}/update`, {
  where: updateSomeProps.id,
  data: {
    ...updateSomeProps,
    someHasManyRelation: [
      { ... }, { ... }
    ]
  },
  insert: ['posts'] // <- "Insert" relationship.
})
```

**Comment by rkingon** [comment]
@kiaking that totally worked. for some reason, I was under the impression `insertOrUpdate` was only for the top level model & wasn't thinking recursively.

highly appreciate vuex orm!

**Comment by kiaking** [comment]
@rkingon Yeah currently the method (`insert`, `update` etc) gets applied through all relationships. Great hear that it worked! 👍
